File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        mid = 0
        r = len(nums)-1
        while l<= r :
            mid = (l+r)//2
            if nums[mid]==target:
                return mid
                
            
            if nums[mid]>target:
                mid-=1
                r = mid
                continue
            mid+=1
            l = mid
            
        
        return -1
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        std::map<int ,int> count;
        vector<vector<int>> freq(nums.size()+1);
       for(int number: nums){
        count[number]++;
       } 
       for(std::pair<int,int> group: count){
        freq[group.second].push_back(group.first);
       }

        
        vector<int> topK;
       for(int frequency = freq.size()-1; frequency>0; frequency--){

            for(int num = 0; num< freq[frequency].size();num++ ){
                topK.push_back(freq[frequency][num]);
                if(topK.size()==k){
                    return topK;
                }
            }
       }
       return topK;
    }
};
